### DISCUSSION TITLE
How to make mitmdump work on a CI pipeline?

### DISCUSSION THREAD

**Discussion by cportuguez-kg** [mitmproxy/mitmproxy#5421] [Q&A]
Hello. We have a mobile project where we use mitmdump to intercept specific API calls so the client app receives different responses and thus we unlock new behavior. We have obtained successful results on local machines for both iOS and Android emulators.

Now, we're trying to obtain the same results on our GitHub Actions pipeline (only iOS for now as a POC) but we're only getting the following error from mitmdump's logs:
`Client TLS handshake failed. The client disconnected during the handshake. If this happens consistently for [our API], this may indicate that the client does not trust the proxy's certificate.`

More context:
- GitHub machine: MacOS just like the ones we have on local.
- Our iOS simulator: We duplicated my iOS sim into the pipeline, the sim works for me locally and we obtain successful results as said before, so since it's the same one we expected to also get good results on CI.
- Certificate: Properly set, same for the profile, just like on the local sim since it's a duplicate. We even ran some dummy jobs to verify the CI sim's settings and everything looks okay.
- mitmproxy installation: We install it with brew on every run.
- Setting the proxy: All set just like on local, we change the Mac's network settings and the iOS sim uses it properly.
- App: Opens properly, but it gets stuck on a certain loading screen, I blame this to the error shared above.

**Comment by rahimrahman** [comment]
@cportuguez-kg did you all every get this to work?

**Comment by mhils** [comment]
mitmproxy generates a unique CA certificate on the first startup so that you cannot be intercepted by other users. It sounds like you are not copying over your mitmproxy CA, so your CI job will generate a new (untrusted) one on each run. You either need to copy over the certificate, or mark the new one as trusted in CI.
